tnb/base_client.py

```diff
@@ -45,27 +45,16 @@ def fetch(self, resource: str, **kwargs: dict) -> Union[dict, list]:
         GET a `resource` from a Node
         Return response as Python object
         """
-        return self.send_request("GET", resource, **kwargs)
-
-    def fetch_multiple_page(self, resource: str, **kwargs) -> list:
-        """
-        GET a complete `resource` from a Node where this resource is splited into
-        multiple pages.
-        Return response as Python object
-        """
-        complete_response = []
-        current_response = self.fetch(resource)
-        complete_response += current_response["results"]
-        next_url = current_response["next"]
+        params = kwargs.get("params", {})
+        page = kwargs.pop("page", 0)
+        limit = kwargs.pop("limit", 0)
+        offset = (page - 1) * limit
 
-        while next_url is not None:
-            current_response = self.send_request_from_url(
-                "GET", current_response["next"]
-            )
-            complete_response += current_response["results"]
-            next_url = current_response["next"]
+        if offset >= 0:
+            params.update({"offset": offset, "limit": limit})
+            kwargs["params"] = params
 
-        return complete_response
+        return self.send_request("GET", resource, **kwargs)
```

Comment:
Wouldn't `self.send_request("GET", resource, params=params)` be better ?

Reply:
No, this will give the developer the opportunity to add additional `requests` arguments via the base `fetch` method e.g. a specific `headers` configuration or `timeout` configuration. All while the client still handles the response logic such as converting the JSON response to a `dictionary` or `list`.

Also to keep it consistent with all the other base methods such as `delete`, `patch`, `post` etc.

Reply:
And to add to the discussion the methods `send_request` and `send_request_from_url` (I don't think this one is used) should actually become 'private' methods by renaming them to `_send_request` and `_send_request_from_url`. But this is a change for another time/issue.